firebase-firestore/src/main/java/com/google/firebase/firestore/core/Transaction.java

```diff
@@ -133,13 +140,14 @@ private Precondition precondition(DocumentKey key) {
    * Returns the precondition for a document if the operation is an update, based on the provided
    * UpdateOptions.
    */
-  private Precondition preconditionForUpdate(DocumentKey key) {
+  private Precondition preconditionForUpdate(DocumentKey key) throws FirebaseFirestoreException {
     @Nullable SnapshotVersion version = this.readVersions.get(key);
     if (version != null && version.equals(SnapshotVersion.NONE)) {
       // The document to update doesn't exist, so fail the transaction.
-      throw new IllegalStateException("Can't update a document that doesn't exist.");
+      throw new FirebaseFirestoreException(
+          "Can't update a document that doesn't exist.", Code.INVALID_ARGUMENT);
```

Comment:
Spreadsheet F2 calls for this to be NOT_FOUND.

Reply:
This should stay as INVALID_ARGUMENT b/c the user is performing a get first in the transaction.